demos/jobs_for_aws/jobs_demo.c

```diff
@@ -0,0 +1,789 @@
+ */
+static uint8_t ucSharedBuffer[ democonfigNETWORK_BUFFER_SIZE ];
+
+/**
+ * @brief Static buffer used to hold MQTT messages being sent and received.
+ */
+static MQTTFixedBuffer_t xBuffer =
+{
+    .pBuffer = ucSharedBuffer,
+    .size    = democonfigNETWORK_BUFFER_SIZE
+};
+
+/**
+ * @brief A global flag which represents whether a job for the "Exit" action
+ * has been received from AWS IoT Jobs service.
+ */
+static BaseType_t xExitActionJobReceived = pdFALSE;
+
+/**
+ * @brief A global flag which represents whether an error was encountered while
+ * executing the demo.
+ *
+ * @note When this flag is set, the demo terminates execution.
+ */
+static BaseType_t xDemoEncounteredError = pdFALSE;
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Converts a string in a Job document to a #JobActionType
+ * value.
+ *
+ * @param[in] pcAction The Job action as a string.
+ * @param[in] xActionLength The length of @p pcAction.
+ *
+ * @return A #JobActionType equivalent to the given string.
+ */
+static JobActionType prvGetAction( const char * pcAction,
+                                   size_t xActionLength );
+
+/**
+ * @brief This example uses the MQTT library of the AWS IoT Device SDK for
+ * Embedded C. This is the prototype of the callback function defined by
+ * that library. It will be invoked whenever the MQTT library receives an
+ * incoming message.
+ *
+ * @param[in] pxMqttContext MQTT context pointer.
+ * @param[in] pxPacketInfo Packet Info pointer for the incoming packet.
+ * @param[in] pxDeserializedInfo Deserialized information from the incoming packet.
+ */
+static void prvEventCallback( MQTTContext_t * pxMqttContext,
+                              MQTTPacketInfo_t * pxPacketInfo,
+                              MQTTDeserializedInfo_t * pxDeserializedInfo );
+
+/**
+ * @brief Process payload from $aws/things<thingName>/jobs/<jobId>/get/accepted topic.
+ *
+ * This handler parses the payload received about the next pending job to identify
+ * the action requested in the job document, and perform the appropriate
+ * action to execute the job.
+ *
+ * @param[in] pPublishInfo Deserialized publish info pointer for the incoming
+ * packet.
+ */
+static void prvStartNextJobHandler( MQTTPublishInfo_t * pxPublishInfo );
+
+/**
+ * @brief Sends an update for a job to the UpdateJobExecution API of the AWS IoT Jobs service.
+ *
+ * @param[in] pcJobId The job ID whose status has to be updated.
+ * @param[in] usJobIdLength The length of the job ID string.
+ * @param[in] pcJobStatusReport The JSON formatted report to send to the AWS IoT Jobs service
+ * to update the status of @p pcJobId.
+ */
+static void prvSendUpdateForJob( char * pcJobId,
+                                 uint16_t usJobIdLength,
+                                 const char * pcJobStatusReport );
+
+/**
+ * @brief Executes a Job received from AWS IoT Jobs service and sends an update back to the service.
```

Comment:
Can you capitalize Job everywhere, if it so here?

Reply:
https://docs.aws.amazon.com/iot/latest/developerguide/jobs-mqtt-api.html#mqtt-nextjobexecutionchanged
The AWS doc uses "job" when using it as a common noun, thus will just stick with use lower case, and change this one to lower case.

Reply:
Ok